Rule/Time/Percentage.ts

```diff
@@ -0,0 +1,26 @@
+import { isly } from "isly"
+import { Base } from "./Base"
+
+export type Percentage = number
+export namespace Percentage {
+	export const type = isly.number()
+	export const is = type.is
+	export const flaw = type.flaw
+	export function parse(time: string): Percentage | undefined {
+		let result: number | undefined
+		const [match, percentageString] = time.match(/^((?:\d)+\.?(?:\d*))%$/) ?? []
+		if (!match)
+			result = undefined
+		else {
+			const percentage = parseFloat(percentageString)
+			if (!Base.safe(percentage))
+				result = undefined
+			else
+				result = percentage / 100
+		}
+		return result
+	}
+	export function toString(time: Percentage): string {
+		return `${(time * 100).toString(10)}%`
```

Comment:
I left a comment elsewhere asking about toString(10) but I found the documentation now and it seems like it is to tell which base to use when writing out the number. Base 10 appears to be the default value so then we wouldnt need to write out 10.

Reply:
It is indeed. I have just preferred to be explicit.